client/src/featureform/register.py

```diff
@@ -1926,6 +1927,58 @@ def register_redis(
         self.__resources.append(provider)
         return OnlineProvider(self, provider)
 
+    def register_pinecone(
+        self,
+        name: str,
+        project_id: str,
+        environment: str,
+        api_key: str,
+        description: str = "",
+        team: str = "",
+        tags: List[str] = [],
+        properties: dict = {},
+    ):
+        """Register a Pinecone provider.
+
+        **Examples**:
+        ```
+        pinecone = ff.register_pinecone(
+            name="pinecone-quickstart",
+            project_id="2g13ek7",
+            environment="us-west4-gcp-free",
+            api_key="e4egd064-1vb6-497f-aadf-7547atbb517f"
+            description="A Pinecone project for we Featureform embeddings"
+        )
+        ```
+
+        Args:
+            name (str): Name of Pinecone provider to be registered
+            project_id (str): Pinecone project id
+            environment (str): Pinecone environment
+            api_key (str): Pinecone api key
+            description (str): Description of Pinecone provider to be registered
+            team (str): Name of team
+            tags (List[str]): Optional grouping mechanism for resources
+            properties (dict): Optional grouping mechanism for resources
+
+        Returns:
+            pinecone (OnlineProvider): Provider
+        """
+        config = PineconeConfig(
+            project_id=project_id, environment=environment, api_key=api_key
```

Comment:
should we default an environment?

Reply:
We _could_ guess the environment, but it might not work. When I signed up on the free tier, it chose `us-west1-gcp-free` as the environment, which I'd guess is based on location, but that's a guess.

<img width="677" alt="Screen Shot 2023-06-09 at 6 48 23 PM" src="https://github.com/featureform/featureform/assets/10430775/73a96dca-f716-4921-8186-5a110fafe091">